apps/server/src/routes/tags/tag-service.ts

```
@@ -0,0 +1,293 @@
+	private buildTagHierarchy(tags: Tag[]): Tag[] {
+		const tagMap = new Map<string, Tag>();
+		const rootTags: Tag[] = [];
+
+		// Create a map of all tags
+		tags.forEach(tag => {
+			tagMap.set(tag.id, { ...tag, children: [] });
+		});
+
+		// Build hierarchy
+		tags.forEach(tag => {
+			const tagWithChildren = tagMap.get(tag.id)!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)